bindings/rust/src/lib.rs

```diff
@@ -755,4 +825,229 @@ mod tests {
+                .expect("valid mutable slice");
+            assert_eq!(mem[0], 0);
+            assert_eq!(mem[1], 0);
+            assert_eq!(mem[2], 0);
+            assert_eq!(mem[3], 0);
+            mem[0] = 42;
+        }
+        unsafe {
+            // Check that const slice matches up.
+            let mem = instance.checked_memory_slice(0, 5).expect("valid slice");
+            assert_eq!(mem[0], 42);
+            assert_eq!(mem[1], 0);
+            assert_eq!(mem[2], 0);
+            assert_eq!(mem[3], 0);
+        }
+        let result = instance
+            .execute("peek", &[TypedValue::U32(0)], 0)
+            .expect("successful execution");
+        assert!(!result.trapped());
+        assert_eq!(
+            result
+                .value()
+                .expect("expected value")
+                .as_u32()
+                .expect("expected u32 result"),
+            42
+        );
+
+        // Remember, by now we have grown memory to two pages.
+
+        // Set memory via safe helper.
+        assert!(instance.memory_set(0, &[]).is_ok());
+        assert!(instance.memory_set(65536 + 65535, &[]).is_ok());
+        assert!(instance.memory_set(65536 + 65536, &[]).is_err());
+        assert!(instance.memory_set(65536 + 65537, &[]).is_err());
+        assert!(instance.memory_set(0, &[0x11, 0x22, 0x33, 0x44]).is_ok());
+        assert!(instance
+            .memory_set(65536 + 65532, &[0x11, 0x22, 0x33, 0x44])
+            .is_ok());
+        assert!(instance
+            .memory_set(65536 + 65533, &[0x11, 0x22, 0x33, 0x44])
+            .is_err());
+        assert!(instance
+            .memory_set(65536 + 65534, &[0x11, 0x22, 0x33, 0x44])
+            .is_err());
+        assert!(instance
+            .memory_set(65536 + 65535, &[0x11, 0x22, 0x33, 0x44])
+            .is_err());
+        assert!(instance
+            .memory_set(65536 + 65536, &[0x11, 0x22, 0x33, 0x44])
+            .is_err());
+        assert!(instance
+            .memory_set(65536 + 65537, &[0x11, 0x22, 0x33, 0x44])
+            .is_err());
+
+        let result = instance
+            .execute("peek", &[TypedValue::U32(0)], 0)
+            .expect("successful execution");
+        assert!(!result.trapped());
+        assert_eq!(
+            result
+                .value()
+                .expect("expected value")
+                .as_u32()
+                .expect("expected u32 result"),
+            0x44332211
+        );
+
+        // Change memory via wasm.
+        let result = instance
+            .execute(
+                "poke",
+                &[TypedValue::U32(0), TypedValue::U32(0x88776655)],
+                0,
+            )
+            .expect("successful execution");
+        assert!(!result.trapped());
+
+        // Read memory via safe helper.
+        let mut dst = Vec::new();
+        dst.resize(65536, 0);
+        // Reading 65536 bytes.
+        assert!(instance.memory_get(0, &mut dst).is_ok());
```

Comment:
Need also to add tests for `memory_get` with empty dst slice and with overruns.

Reply:
```rust
let mut dst_emtpy = Vec::new();
assert!(instance.memory_get(0, &mut dst).is_ok());
assert!(instance.memory_get(65536 + 65535, &mut dst).is_ok());
assert!(instance.memory_get(65536 + 65536, &mut dst).is_err());
assert!(instance.memory_get(65536 + 65537, &mut dst).is_err());

let mut dst_small = Vec::new();
dst_small.resize(4, 0);
assert!(instance.memory_get(0, &mut dst_small).is_ok());
assert!(instance
            .memory_get(65536 + 65532, &mut dst_small)
            .is_ok());
assert!(instance
            .memory_get(65536 + 65533, &mut dst_small)
            .is_err());
assert!(instance
            .memory_get(65536 + 65534, &mut dst_small)
            .is_err());
assert!(instance
            .memory_get(65536 + 65535, &mut dst_small)
            .is_err());
assert!(instance
            .memory_get(65536 + 65536, &mut dst_small)
            .is_err());
assert!(instance
            .memory_get(65536 + 65537, &mut dst_small)
            .is_err());
```

Reply:
Added, but using slices.